### PR TITLE
Minor tweaks to Effect's API for consistency

### PR DESCRIPTION
**Pull request by stephencelis**
A few things I noticed. Up for workshopping further.

### CODE REVIEW

File: Sources/ComposableArchitecture/Effect.swift [hunk 1]
```diff
@@ -89,14 +89,14 @@ public struct Effect<Output, Failure: Error>: Publisher {
   ///  If you need to deliver more than one value to the effect, you should use the `Effect`
   ///  initializer that accepts a `Subscriber` value.
   ///
-  /// - Parameter work: A closure that takes a `callback` as an argument which can be used to feed
-  ///   it `Output` values.
+  /// - Parameter attemptToFulfill: A closure that takes a `callback` as an argument which can be
```
Comment (stephencelis) [inline Sources/ComposableArchitecture/Effect.swift:92]: This matches Combine's API.

File: Sources/ComposableArchitecture/Effect.swift [hunk 2]
```diff
@@ -158,13 +158,13 @@ public struct Effect<Output, Failure: Error>: Publisher {
...
   ///
-  /// - Parameter run: A closure that accepts a `Subscriber` value and returns a cancellable. When
-  ///   the `Effect` is completed, the cancellable will be used to clean up any
-  ///   resources created when the effect was started.
+  /// - Parameter work: A closure that accepts a `Subscriber` value and returns a cancellable. When
+  ///   the `Effect` is completed, the cancellable will be used to clean up any resources created
+  ///   when the effect was started.
   public static func async(
-    _ run: @escaping (Effect.Subscriber<Output, Failure>) -> Cancellable
+    _ work: @escaping (Effect.Subscriber<Output, Failure>) -> Cancellable
```
Comment (stephencelis) [inline Sources/ComposableArchitecture/Effect.swift:165]: We use `run` some places and `work` other places. In fact I wonder if this entire thing could be renamed from `async` to `run` 🤔

`Effect.run { subscriber in ... }` has a nice ring to it...

Comment (mbrandonw) [reply Sources/ComposableArchitecture/Effect.swift:165]: Dang, yeah. That’s pretty nice

Comment (stephencelis) [reply Sources/ComposableArchitecture/Effect.swift:165]: Should we make the change in this PR? Separate? Deprecate with `renamed:`?

Comment (stephencelis) [reply Sources/ComposableArchitecture/Effect.swift:165]: I'll merge as is. We can think about this a bit more.

**Pull request merged**